### DISCUSSION TITLE
refusing to allow an OAuth App to create or update workflow '.github/workflows/ci.yml' without workflow scope

### DISCUSSION THREAD

**Discussion by bradzacher** [community/community#45947] [Mobile] [answered]
### Select Topic Area

Bug

### Body

Today I tried to merge a PR that made a change to a workflow file from. The GH mobile app.

I got the following error toast

> refusing to allow an OAuth App to create or update workflow '.github/workflows/ci.yml' without workflow scope

![Screenshot_20230201-063524](https://user-images.githubusercontent.com/7462525/215873590-b8f95cf1-e956-4ac0-a508-322acc597c88.png)

The app provided no steps to remedy the situation - it just blocked me completely with a toast - which is a terrible UX!

Googling around I found various people that said to [update the PAT with the workflow scope](https://stackoverflow.com/questions/64059610/how-to-resolve-refusing-to-allow-an-oauth-app-to-create-or-update-workflow-on), but I can't find the PAT I used for GH mobile. I tried adding it to all my PATs, then cleared the app storage and logged in again, but that didn't fix it.

I don't rember using a PAT because it was an OAuth App, right? I checked the various settings pages for my account and could find no reference to the GH mobile session aside from the active sessions page.

I tried reinstalling the app afresh to nuke the storage from orbit, and it didn't fix it.

So I don't know how to fix my app so that I can merge PRs that touch workflpw files. For now I've had to jump back out of the app to the mobile website.

**Comment by jge162** [comment]
Hello @bradzacher sorry to hear you are having issues. Try these steps to see if it remedies your issues.

- Go to your GitHub settings page on the desktop website
- Select Developer settings
- Select Personal access tokens
- Create a new PAT with the "workflow" scope selected
- Use this new PAT to log in to the GitHub mobile app

**Reply by jge162** [reply]
and if does not work consider reaching out to [GitHub Support](https://support.github.com/request/account?tags=hubberfy_account)

**Reply by bradzacher** [reply]
> Use this new PAT to log in to the GitHub mobile app

How do I do this?
When I login to the app I don't get an option to select anything in particular.
It just automatically goes through the browser - there are no prompts to select a specific PAT.

**Reply by jge162** [reply]
You should create a support ticket for more assistance.

**Comment by alcere** [comment, marked as answer]
Hi @bradzacher thanks for reporting! We are rolling out a fix for this, we will let you know once this is fixed, or you can try re-logging into the app if you are on the latest beta, we are sorry for the troubles!

**Reply by making** [reply]
@alcere 
I also wanted to create a workflow from an Oauth app and ran into the above issue. Multiple sites state that the workaround for this issue is to use PAT, but our function cannot accept PAT.
Has this issue been fixed? Or is it a "specification" that a workflow cannot be created from an Oauth app?

**Reply by alcere** [reply]
Hi @making not sure if I've understood the issue, are you using a different app than mobile? This is caused by a security limitation imposed by the OAuth scopes granted to the OAuth user token, workflow files cannot be modified without the `workflow` [scope](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes).

**Reply by making** [reply]
@alcere That's my bad. It was our app's issue.

**Comment by spaltrowitz** [comment]
Hi @bradzacher @making, thanks for your feedback. A feature related to your feedback has now shipped! 🚢 https://github.blog/changelog/2023-05-11-edit-workflow-files-on-github-mobile/ cc @alcere
